### PR TITLE
fix(core): decode ~0 tilde escapes in JSON Pointer refs

### LINKED ISSUE #3413: fix(core): decode ~0 tilde escapes in JSON Pointer refs
## Problem

`getRefInfo` in `packages/core/src/getters/ref.ts` decodes JSON Pointer path segments for `$ref` values, but it only handles the `/` escape (`~1`). It does not decode the tilde escape (`~0`).

Per [RFC 6901 section 3](https://datatracker.ietf.org/doc/html/rfc6901#section-3), JSON Pointer reference tokens must decode both escape sequences:

- `~1` becomes `/`
- `~0` becomes `~`

The replacement order matters. `~1` should be decoded before `~0` so escaped tokens are not decoded incorrectly.

## Current behavior

A schema named `My~Type` must be referenced as:

```json
{ "$ref": "#/components/schemas/My~0Type" }
```

Today, Orval leaves the escaped tilde in the decoded path segment. That means `#/components/schemas/My~0Type` is treated as if the schema name were `My~0Type`, instead of `My~Type`.

This can cause refs to resolve to the wrong name or fail to resolve when a component key contains a literal `~`.

## Expected behavior

`getRefInfo` should decode JSON Pointer escape sequences according to RFC 6901:

```ts
part.replaceAll('~1', '/').replaceAll('~0', '~')
```

For example:

| Ref | Decoded schema name |
| --- | --- |
| `#/components/schemas/My~0Type` | `My~Type` |
| `#/components/schemas/Path~1To~0Thing` | `Path/To~Thing` |

## Impact

This is a general `$ref` correctness issue. Specs with component names containing a literal tilde (`~`) can generate incorrect names or fail to resolve refs correctly.

The fix should only affect refs that contain `~0`; existing refs without tilde escapes should continue to behave the same.

## Suggested scope

Files likely involved:

- `packages/core/src/getters/ref.ts`
- `packages/core/src/getters/ref.test.ts`
- `packages/core/src/resolvers/ref.test.ts`

Suggested test coverage:

- `getRefInfo` decodes `#/components/schemas/My~0Type` to `My~Type`
- `getRefInfo` decodes a path segment containing both `~1` and `~0`
- `resolveRef` resolves a schema whose component key contains a literal `~`

### PR DESCRIPTION
**Pull request by aqeelat**
## Summary

`getRefInfo` in `packages/core/src/getters/ref.ts` now decodes both RFC 6901 JSON Pointer escape sequences in the correct order:

1. `~1` → `/`
2. `~0` → `~`

Previously only `~1` was decoded, leaving `~0` as a literal in the decoded path segment. This caused `$ref` values like `#/components/schemas/My~0Type` to resolve to `My~0Type` instead of `My~Type`.

## Test plan

- `getRefInfo` decodes `#/components/schemas/My~0Type` → `My~Type`
- `getRefInfo` decodes a segment containing both `~1` and `~0` (`Path~1To~0Thing` → `Path/To~Thing`)
- `getRefInfo` decodes `~01` → `~1` (validates that `~1` is replaced before `~0`)
- `resolveRef` resolves a schema whose component key contains a literal `~`

Closes #3413

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected reference decoding so component names with tildes (~) and JSON Pointer escape sequences are handled reliably (including proper percent-decoding and tilde unescaping).

* **Tests**
  * Added tests covering JSON Pointer unescaping edge cases and resolving refs that include literal tildes to prevent regressions.

[Review Change Stack](https://app.coderabbit.ai/change-stack/orval-labs/orval/pull/3414?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/core/src/getters/ref.test.ts [hunk 1]
```diff
@@ -40,3 +56,30 @@ describe('isComponentRef', () => {
...
 });
+
+describe('getRefInfo', () => {
+  it('decodes ~0 tilde escape per RFC 6901', () => {
+    const info = getRefInfo(
+      '#/components/schemas/My~0Type',
+      createRefContext(),
+    );
+    expect(info.originalName).toBe('My~Type');
+  });
```
[the remark this answers is not in the document]
Comment (aqeelat) [reply packages/core/src/getters/ref.test.ts:67]: added.

File: packages/core/src/getters/ref.test.ts [hunk 2]
```diff
@@ -40,3 +56,42 @@ describe('isComponentRef', () => {
...
```
Comment (aqeelat) [inline packages/core/src/getters/ref.test.ts:67]: this test is partially covered by the two tests below. not sure if we should keep it for explicit scenario coverage or not.

**Review by melloware — APPROVED** [review]
